engine/src/main/java/io/zeebe/engine/processing/bpmn/behavior/BpmnEventSubscriptionBehavior.java

```diff
@@ -456,9 +457,13 @@ public void publishTriggeredEventSubProcess(final BpmnElementContext context) {
     }
   }
 
-  private boolean isInterrupted(final ElementInstance elementInstance) {
-    return elementInstance.getNumberOfActiveElementInstances() == 1
-        && elementInstance.isInterrupted()
-        && elementInstance.isActive();
+  private boolean isInterrupted(
+      final boolean isChildMigrated, final ElementInstance elementInstance) {
+    final boolean shouldITurnOffTheLights =
+        isChildMigrated
+            ? elementInstance.getNumberOfActiveElementInstances() == 0
+            : elementInstance.getNumberOfActiveElementInstances() == 1;
+
+    return shouldITurnOffTheLights && elementInstance.isInterrupted() && elementInstance.isActive();
```

Comment:
:smile: great name

How about:

```suggestion
final int expectedActiveInstanceCount = isChildMigrated ? 0 : 1;
    return elementInstance.getNumberOfActiveElementInstances() == expectedActiveInstanceCount
        && elementInstance.isInterrupted()
        && elementInstance.isActive();
```

That way later we can just inline the variable and remove it.

Reply:
Haha lol actually it did this when I discussed with phil the solution but have forgotten to change the name again :laughing: